fetchai/ledger/crypto/deed.py

```diff
@@ -0,0 +1,66 @@
+# ------------------------------------------------------------------------------
+#
+#   Copyright 2018-2019 Fetch.AI Limited
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+# ------------------------------------------------------------------------------
+import logging
+from typing import Union
+
+from fetchai.ledger.crypto import Address, Identity, Entity
+
+AddressLike = Union[Address, Identity]
+
+class Deed:
+    def __init__(self, address: Entity):
+        self._address = address
+        self._amend_threshold = None
+        self._transfer_threshold = None
+        self._signees = {}
+
+    def add_signee(self, signee: Entity, voting_weight: int):
+        self._signees[signee] = voting_weight
+
+    @property
+    def amend_threshold(self):
+        return self._amend_threshold
+
+    @amend_threshold.setter
+    def amend_threshold(self, value):
+        self._amend_threshold = value
+
+    @property
+    def transfer_threshold(self):
+        return self._transfer_threshold
+
+    @transfer_threshold.setter
+    def transfer_threshold(self, value):
+        self._transfer_threshold = value
+
+    def deed_creation_json(self):
+        deed = {
+            'address': Address(self._address)._display,
```

Comment:
The `address` field can be dropped, it is the legacy left-over we forgot to ditch in C++ ledger side when migrating from to the new Transaction serialisation format V2.
I just making PR on C++ ledger side to ditch the requirement for presence of the field in json deed expected to be provided in `data` field of Transaction carrying the deed.

Reply:
Removing this field from the json causes the deed creation transaction to fail.

Reply:
The above-mentioned PR in ledger is dedicated sort that out.

Reply:
Ah, sorry - hadn't realised the other PR hadn't landed yet. Will leave as is for now.